sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorConnection.java

```diff
@@ -135,9 +134,9 @@ public ReactorConnection(String connectionId, ConnectionOptions connectionOption
                 if (!isDisposed.getAndSet(true)) {
                     logger.verbose("connectionId[{}]: Disposing of active sessions due to error.", connectionId);
                     return dispose(new AmqpShutdownSignal(false, false,
-                        error.getMessage())).then(Mono.empty());
+                        error.getMessage())).then(Mono.error(error));
                 } else {
-                    return Mono.empty();
+                    return Mono.error(error);
```

Comment:
Pass the transient error to downstream so it creates a new connection

Reply:
This looks good.